work_queue/src/bindings/python3/brett.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+import work_queue as wq
+
+def wqmap(f, a):
+    q = wq.WorkQueue(9123)
+    results = []
+    for i in a:
+        p_task = wq.PythonTask(f, i)
+        q.submit(p_task)
+
+    while not q.empty():
+        t = q.wait()
+        if t:
+            x = t.output
+            if isinstance(x, wq.PythonTaskNoResult):
+                print("Task {} failed and did not generate a result.".format(t.id))
+            else:
+                results.insert(0, x)
+
+    return results
+
+def wqallpairs(f, a, b):
+    q = wq.WorkQueue(9123)
+    results = []
+    for i in a:
```

Comment:
We can use here:

```python
import itertools

for i, (x,y) in enumerate(itertools.product(a, b)):
    p_task = wq.PythonTask(f, x, y)
   ... keep a dictionary from p_task to i so that we know the order of submission.
  ... and when inserting to results, we do it in a predictable (i.e., itertools.product) order.
```

As with `map`, all these should be methods of WorkQueue, thus no need to create a queue here and worry about the port number.

Reply:
Ok that makes a lot of sense.

Reply:
@btovar What do you mean here by "in a predictable (i.e., itertools.product) order"?